metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1995,6 +2015,19 @@ private void listReassigningTopic(ListPartitionReassignmentsResponseData respons
             setReplicas(Replicas.toList(partition.replicas)));
     }
 
+    // Visible to test.
+    Integer getTopicEffectiveMinIsr(String topicName) {
+        String minIsrConfig = configurationControl.getTopicConfig(topicName, MIN_IN_SYNC_REPLICAS_CONFIG);
+        Integer currentMinIsr = minIsrConfig.isEmpty() ? defaultMinIsr : Integer.parseInt(minIsrConfig);
+        Uuid topicId = topicsByName.get(topicName);
+        Integer replicationFactor = topics.get(topicId).parts.get(0).replicas.length;
```

Comment:
Yeah, from the current use case, all the get will have valid values. However, I think we can be paranoid here to have a try-catch to avoid mistakes as there are a lot of places can hit null. 
@mumrah You commented on this before, what do you think?

Reply:
Updated.